collated/main/A0148096W.md

```diff
@@ -0,0 +1,1230 @@
+        // No arguments, use default 'list' command
+        if (args.trim().equals("")) {
+            try {
+                return new ListCommand();
+            } catch (IllegalValueException ive) {
+                return new IncorrectCommand(ive.getMessage());
+            }
+        }
+
+        final KeywordParser parser = new KeywordParser(ListCommand.VALID_KEYWORDS);
+        HashMap<String, String> parsed = parser.parseKeywordsWithoutFixedOrder(ListCommand.COMMAND_WORD + args);
+        String type = setToNullIfIsEmptyString(parsed.get(ListCommand.COMMAND_WORD));
+        String deadline = setToNullIfIsEmptyString(parsed.get(ListCommand.KEYWORD_DEADLINE));
+        String startTime = setToNullIfIsEmptyString(parsed.get(ListCommand.KEYWORD_PERIOD_START_TIME));
+        String endTime = setToNullIfIsEmptyString(parsed.get(ListCommand.KEYWORD_PERIOD_END_TIME));
+        String tags = setToNullIfIsEmptyString(parsed.get(ListCommand.KEYWORD_TAG));
+        String sortingOrder = setToNullIfIsEmptyString(parsed.get(ListCommand.KEYWORD_SORT));
+
+        if(tags == null){
+            tags = "";
+        } else {
+            tags = removeFullStopsAndCommas(tags);
+        }
+
+        if (sortingOrder != null) {
+            sortingOrder = removeFullStopsAndCommas(sortingOrder);
+        }
+
+        try {
+            return new ListCommand(
+                    type,
+                    deadline,
+                    startTime,
+                    endTime,
+                    getTagsFromArgs(tags),
+                    sortingOrder
+            );
+        } catch (IllegalValueException ive) {
+            return new IncorrectCommand(ive.getMessage());
+        }
+    }
+
+    /**
+     * Precondition: argument string is not null.
+     * Parses the command string in the context of the show command.
+     *
+     * @param args full command args string
+     * @return the prepared command
+     */
+    private Command prepareShow(String args){
+        assert args != null;
+
+        if (args.trim().equals("")) {
+            return new IncorrectCommand(String.format(MESSAGE_INVALID_COMMAND_FORMAT, ShowCommand.MESSAGE_USAGE));
+        }
+
+        final KeywordParser parser = new KeywordParser(ShowCommand.VALID_KEYWORDS);
+        HashMap<String, String> parsed = parser.parseKeywordsWithoutFixedOrder(ShowCommand.COMMAND_WORD + args);
+        String type = setToNullIfIsEmptyString(parsed.get(ShowCommand.COMMAND_WORD));
+        String date = setToNullIfIsEmptyString(parsed.get(ShowCommand.KEYWORD_DATE));
+        String deadline = setToNullIfIsEmptyString(parsed.get(ShowCommand.KEYWORD_DEADLINE));
+        String startTime = setToNullIfIsEmptyString(parsed.get(ShowCommand.KEYWORD_PERIOD_START_TIME));
+        String endTime = setToNullIfIsEmptyString(parsed.get(ShowCommand.KEYWORD_PERIOD_END_TIME));
+        String tags = setToNullIfIsEmptyString(parsed.get(ShowCommand.KEYWORD_TAG));
+
+        if(tags == null){
+            tags = "";
```

Comment:
aren't these few lines duplicated in many methods? can you think of extracting them as a method by itself?